### PR TITLE
[DOCS] Refactor rollover API docs

### PR DESCRIPTION
**Pull request by jrodewig**
Refactors the rollover API docs to focus on data streams.

Changes:
- Reuses the `conditions` definitions from the ILM `rollover` action docs
- Adds response body definitions
- Notes that several parameters do not support data streams
- Separates examples based on rollover target

### Preview
https://elasticsearch_70938.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/indices-rollover-index.html

### CODE REVIEW

File: docs/reference/indices/rollover-index.asciidoc [hunk 1]
```diff
@@ -43,530 +30,396 @@ POST /alias1/_rollover/my-index-000002
...
 
-[source,console]
------------------------------------
-PUT _index_template/template
-{
-  "index_patterns": ["my-data-stream*"],
-  "data_stream": { }
-}
------------------------------------
+* The index was created 7 or more days ago.
```
Comment (danhermann) [inline docs/reference/indices/rollover-index.asciidoc:223]: There's no description of the `max_primary_shard_size` option. We might also want to remove the `max_size` option as it is being deprecated in favor of the new `max_primary_shard_size` option.

**Review by danhermann — COMMENTED** [review]
Added a couple minor comments below.

File: docs/reference/indices/rollover-index.asciidoc [hunk 2]
```diff
@@ -43,530 +30,396 @@ POST /alias1/_rollover/my-index-000002
...
+The API's behavior depends on the rollover target.
+
+**Roll over a data stream**
 
-NOTE: To roll over an index, a condition must be met *when you call the API*.
-{es} does not monitor the index after you receive an API response. To
-automatically roll over indices when a condition is met, you can use {es}'s
-<<index-lifecycle-management, index lifecycle management (ILM) policies>>.
+If you roll over a data stream, the API creates a new write index for the
+stream. The stream's previous write index becomes a read-only backing index. A
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/indices/rollover-index.asciidoc:42]: Good point. I'll reword this to "regular backing index."

File: docs/reference/indices/rollover-index.asciidoc [hunk 3]
```diff
@@ -43,530 +30,396 @@ POST /alias1/_rollover/my-index-000002
...
+Name of the data stream or index alias to roll over.
 
 `<target-index>`::
-+
---
 (Optional*, string)
-Name of the target index to create and assign the index alias.
-
+Name of the index to create. Supports <<date-math-index-names,date math>>. Data
+streams do not support his parameter.
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/indices/rollover-index.asciidoc:103]: Typically, the star convention indicates that a parameter is required in some circumstances but not others. I'll remove it here. In most cases, the user won't provide this parameter.

File: docs/reference/indices/rollover-index.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/indices/rollover-index.asciidoc:223]: I reuse the `conditions` options from the ILM rollover action. It's not easily viewed in the source, but you can see the section in the preview:

https://elasticsearch_70938.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/indices-rollover-index.html#rollover-index-api-request-body

Screen Shot 2021-03-30 at 9 58 07 AM

Comment (jrodewig) [reply docs/reference/indices/rollover-index.asciidoc:223]: I'll also leave in the `max_size` option for now. When we officially deprecate it, I'll add the appropriate admon.

**Comment by jrodewig** [comment]
Thanks @danhermann! I addressed some of your feedback with 8dde9dd08d955de7a4d8c8bbbd3353df7429ad62. Let me know if you have any other thoughts!

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
Hi @danhermann! You mind giving this one more look when convenient? I'd love to go ahead and get these changes merged in.

Thanks!

**Review by danhermann — APPROVED** [review]
Yes, sorry, didn't mean for any of my comments above to be blocking. Everything LGTM.

**Comment by jrodewig** [comment]
No worries at all. Thanks again @danhermann.

**Review by andreidan — APPROVED** [review]
Sorry for the late review @jrodewig 

Thanks for working on this! LGTM
